diff --git a/node_modules/@whitespace/storybook-addon-code/dist/SyntaxHighlighter.js b/node_modules/@whitespace/storybook-addon-code/dist/SyntaxHighlighter.js
index f7d038f..39f2282 100644
--- a/node_modules/@whitespace/storybook-addon-code/dist/SyntaxHighlighter.js
+++ b/node_modules/@whitespace/storybook-addon-code/dist/SyntaxHighlighter.js
@@ -9,7 +9,7 @@ var _react = _interopRequireWildcard(require("react"));
 
 var _theming = require("@storybook/theming");
 
-var _ActionBar = require("@storybook/components/dist/ActionBar/ActionBar");
+var _ActionBar = require("@storybook/components/dist/esm/ActionBar/ActionBar");
 
 var _reactSyntaxHighlighter = _interopRequireDefault(require("react-syntax-highlighter"));
 
