diff --git a/src/index.js b/src/index.js
index 3197dc8..406abfe 100644
--- a/node_modules/drupal-attribute/src/index.js
+++ b/node_modules/drupal-attribute/src/index.js
@@ -100,6 +100,44 @@ class DrupalAttribute extends Map {
 
     return result;
   }
+
+  /**
+   * Merge another DrupalAttribute instance into this one.
+   * @param {DrupalAttribute} collection
+   * @returns {DrupalAttribute}
+  */
+  merge(collection) {
+    // Convert both the current attributes and the input collection's attributes to plain objects.
+    let currentAttributes = Object.fromEntries(this);
+    let newAttributes = Object.fromEntries(collection);
+
+    // Perform a deep merge.
+    let mergedAttributes = this.mergeDeep(currentAttributes, newAttributes);
+
+    // Clear the current Map and set the merged attributes.
+    this.clear();
+    for (let key in mergedAttributes) {
+      this.set(key, mergedAttributes[key]);
+    }
+
+    return this;
+  }
+  
+  // Helper function for deep merging.
+  mergeDeep(target, source) {
+    for (let key in source) {
+      if (source[key] instanceof Array) {
+        if (!target[key]) target[key] = [];
+        target[key] = Array.from(new Set(target[key].concat(source[key])));
+      } else if (source[key] instanceof Object) {
+        if (!target[key]) target[key] = {};
+        Object.assign(source[key], this.mergeDeep(target[key], source[key]));
+      } else {
+        target[key] = source[key];
+      }
+    }
+    return target;
+  }
 }
 
-module.exports = DrupalAttribute;
\ No newline at end of file
+module.exports = DrupalAttribute;
